### PR TITLE
feat: add bar plot

### PR DESCRIPTION
**Pull request by linsamtw**
1. add bar plot
2. add BarPlotSchema
3. kline add filename param

```python
from FinMind import plotting
from FinMind.data import DataLoader
from FinMind.schema import BarPlotSchema

data_loader = DataLoader()
df = data_loader.taiwan_stock_month_revenue(
    stock_id='2330', start_date='2020-1M', end_date='2021-7M'
)
df["labels"] = (
    df[["revenue_year", "revenue_month"]]
    .astype(str)
    .agg(lambda date: f"{date[0]}-{date[1]}M", axis=1)
)
df["series"] = df["revenue"].agg(lambda value: round(value / 1e8, 2))
bar_plot_data = BarPlotSchema.df_convert(df)

plotting.bar(bar_plot_data, title='月營收', yaxis_color='orange')
```
image

```python
plotting.bar?

Signature:
plotting.bar(
    bar_plot_data:FinMind.schema.plot.BarPlotSchema,
    y_series_name:str='',
    y_axis_name:str='億',
    yaxis_color:str='#dca540',
    title:str='title',
    width:str='800px',
    height:str='600px',
    filename:str='bar.html',
)
Docstring:
plot bar
:param: bar_plot_data (:obj:FinMind.BarPlotSchema)
BarPlotSchema(labels=labels, series=series)
:param: y_series_name (str) default ''
:param: y_axis_name (str) default "億",
:param: yaxis_color (str) default "#dca540",
:param: title (str) default "title",
:param: width (str) default "800px",
:param: height (str) default "600px",
:param: filename (str) default "bar.html", output filename

:return: display bar
:rtype pyecharts.charts.Bar
File:      ~/FinMind/FinMind/plotting/bar.py
Type:      function
```

### CODE REVIEW

File: tests/plotting/test_bar.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@pytest.mark.parametrize("stock_id, start_date, end_date", testdata_bar)
+def test_bar(stock_id, start_date, end_date):
+    user_id = os.environ.get("FINMIND_USER", "")
+    password = os.environ.get("FINMIND_PASSWORD", "")
+    data_loader = DataLoader()
+    data_loader.login(user_id, password)
+    df = data_loader.taiwan_stock_month_revenue(
+        stock_id=stock_id, start_date=start_date, end_date=end_date
+    )
+    df["labels"] = (
```
Comment (machineCYC) [inline tests/plotting/test_bar.py:23]: df["labels"] = df["revenue_year"].astype(str) + '-' + df["revenue_month"].astype(str) + 'M'
這樣會不會比較直觀一點?

File: tests/plotting/test_bar.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        df[["revenue_year", "revenue_month"]]
+        .astype(str)
+        .agg(lambda date: f"{date[0]}-{date[1]}M", axis=1)
+    )
+    df["series"] = df["revenue"].agg(lambda value: round(value / 1e8, 2))
```
Comment (machineCYC) [inline tests/plotting/test_bar.py:28]: df["series"] = df["revenue"].map(lambda value: round(value / 1e8, 2))

對單個欄位做operator 是不是用map比較好啊? agg 印象中要跟agg function 用比較合適吧?!

File: tests/schema/test_plot.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    data_loader.login(user_id, password)
+    df = data_loader.taiwan_stock_month_revenue(
+        stock_id="2890", start_date="2018-1M", end_date="2021-7M"
+    )
+    df["labels"] = (
+        df[["revenue_year", "revenue_month"]]
+        .astype(str)
+        .agg(lambda date: f"{date[0]}-{date[1]}M", axis=1)
+    )
+    df["series"] = df["revenue"].agg(lambda value: round(value / 1e8, 2))
```
Comment (machineCYC) [inline tests/schema/test_plot.py:23]: 同上問題

File: tests/schema/test_plot.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+@pytest.fixture(scope="module")
+def df():
+    user_id = os.environ.get("FINMIND_USER", "")
+    password = os.environ.get("FINMIND_PASSWORD", "")
+    data_loader = DataLoader()
+    data_loader.login(user_id, password)
+    df = data_loader.taiwan_stock_month_revenue(
+        stock_id="2890", start_date="2018-1M", end_date="2021-7M"
+    )
+    df["labels"] = (
```
Comment (machineCYC) [inline tests/schema/test_plot.py:18]: 同上問題

**Review by machineCYC — APPROVED** [review]
參考看看!

File: tests/plotting/test_bar.py (hunk 1 above)
Comment (linsamtw) [reply tests/plotting/test_bar.py:23]: 不太喜歡用 + 這種寫法

File: tests/plotting/test_bar.py (hunk 2 above)
Comment (linsamtw) [reply tests/plotting/test_bar.py]: 改囉

File: tests/schema/test_plot.py (hunk 3 above)
Comment (linsamtw) [reply tests/schema/test_plot.py]: 改囉

File: tests/plotting/test_bar.py (hunk 1 above)
Comment (machineCYC) [reply tests/plotting/test_bar.py:23]: 用 agg 真的有點怪! 應該是可以用apply?

**Review by machineCYC — APPROVED** [review]

File: tests/plotting/test_bar.py (hunk 1 above)
Comment (Featasy) [reply tests/plotting/test_bar.py:23]: 其實我覺得還ok耶
如果不是理解上容易產生歧義
或是有什麼重大的效能問題
我們在code review就尊重原作者的想法吧d(`･∀･)b

Comment (linsamtw) [reply tests/plotting/test_bar.py:23]: 已改成 apply

File: tests/schema/test_plot.py (hunk 4 above)
Comment (linsamtw) [reply tests/schema/test_plot.py:18]: 已改成 apply

File: tests/plotting/test_bar.py (hunk 1 above)
Comment (machineCYC) [reply tests/plotting/test_bar.py:23]: @Featasy 主要是因為agg 應該是需要搭配agg funciton ex: min, max...等，會比較正確

File: FinMind/schema/plot.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        super(BarPlotSchema, self).__init__(
+            **{
+                **dict(
+                    labels=labels,
+                    series=series,
+                ),
+                **kwargs,
+            }
+        )
+
```
Comment (Featasy) [inline FinMind/schema/plot.py:25]:
```
        super(BarPlotSchema, self).__init__(
            **{
                **dict(
                    labels=labels,
                    series=series,
                ),
                **kwargs,
            }
        )
 ```
 =>
 ```
         super(BarPlotSchema, self).__init__(
            labels=labels,
            series=series,
            **kwargs,
        )
 ```

File: tests/schema/test_plot.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        df[["revenue_year", "revenue_month"]]
+        .astype(str)
+        .apply(lambda date: f"{date[0]}-{date[1]}M", axis=1)
+    )
+    df["series"] = df["revenue"].map(lambda value: round(value / 1e8, 2))
```
Comment (Featasy) [inline tests/schema/test_plot.py:23]: 建議使用 value * 1e-8

File: FinMind/schema/plot.py (hunk 5 above)
Comment (linsamtw) [reply FinMind/schema/plot.py:21]: 已修正

File: tests/schema/test_plot.py (hunk 6 above)
Comment (linsamtw) [reply tests/schema/test_plot.py]: 已修正

File: FinMind/schema/plot.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        return value
+
+
+def check_labels_series_schema(
+    labels: typing.Union[typing.List[typing.Union[str, int]], Labels],
+    series: typing.Union[typing.List[typing.Union[int, float]], Series],
+):
+    if isinstance(labels, Labels):
+        pass
+    elif isinstance(labels, pd.core.series.Series):
```
Comment (Featasy) [inline FinMind/schema/plot.py:45]: pd.core.series.Series  => pd.Series

File: FinMind/schema/plot.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        labels = list(labels)
+        labels = Labels(labels=labels)
```
Comment (Featasy) [inline FinMind/schema/plot.py:47]:
```
        labels = list(labels)
        labels = Labels(labels=labels)
```
=>
```
        labels = Labels(labels=labels.tolist())
```

File: FinMind/schema/plot.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        elif isinstance(value, float):
+            pass
+        else:
+            raise Exception(
+                "the type of series must be list[int] for list[str]."
+            )
+        return value
+
+
+def check_labels_series_schema(
```
Comment (Featasy) [inline FinMind/schema/plot.py:39]: check 要改成 convert 嗎？  因為裡面其實是在做 convert 的動作

File: FinMind/schema/plot.py [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    @validator("labels", allow_reuse=True, each_item=True)
+    def valid(cls, value):
+        if isinstance(value, str):
+            pass
+        elif isinstance(value, int):
+            pass
+        else:
+            raise Exception(
+                "the type of labels must be list[int] for list[str]."
+            )
```
Comment (Featasy) [inline FinMind/schema/plot.py:19]: 這樣不知道會不會比較好
```
if not isinstance(value, (str, int)):
            raise Exception(
                "the type of labels must be list[int] for list[str]."
            )

**Review by Featasy — COMMENTED** [review]
PTAL

**Review by Featasy — APPROVED** [review]
LGTM

File: FinMind/schema/plot.py (hunk 9 above)
Comment (linsamtw) [reply FinMind/schema/plot.py]: 已修正

File: FinMind/schema/plot.py (hunk 8 above)
Comment (linsamtw) [reply FinMind/schema/plot.py]: 已修正

File: FinMind/schema/plot.py (hunk 7 above)
Comment (linsamtw) [reply FinMind/schema/plot.py]: 已修正

File: FinMind/schema/plot.py (hunk 10 above)
Comment (linsamtw) [reply FinMind/schema/plot.py:19]: 覺得有點多此一舉，所以沒改
前面已經判斷 if str, elif int, 
最後 else 就是 not isinstance(value, (str, int))
